main.tf

```diff
@@ -181,3 +181,33 @@ resource "aws_wafv2_web_acl_association" "this" {
   resource_arn = aws_api_gateway_stage.this[count.index].arn
   web_acl_arn  = var.waf_acl
 }
+
+resource "aws_apigatewayv2_domain_name" "this" {
+  count = var.create_api_domain_name && length(var.stage_names) > 0 ? length(var.stage_names) : 0
+
+  domain_name = var.domain_names[count.index]
+
+  domain_name_configuration {
+    certificate_arn                        = var.domain_certificate_arn
+    ownership_verification_certificate_arn = var.domain_ownership_verification_certificate_arn
+    endpoint_type                          = "REGIONAL"
+    security_policy                        = "TLS_1_2"
```

Comment:
Do we want make variables for these or just tell users they can override the values?

```
endpoint_type  = "REGIONAL"
security_policy = "TLS_1_2"
```

Reply:
I see. The difference here is that v2 does not support `EDGE` but v1 does. We should use the v1 resource so people can make a choice.

```
[types](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_domain_name#types) - (Required) List of endpoint types. This resource currently only supports managing a single value. Valid values: EDGE or REGIONAL. If unspecified, defaults to EDGE. Must be declared as REGIONAL in non-Commercial partitions. Refer to the [documentation](https://docs.aws.amazon.com/apigateway/latest/developerguide/create-regional-api.html) for more information on the difference between edge-optimized and regional APIs.
```

Reply:
moved to use v1 resources. Thanks!